### PR TITLE
ERR: MultiIndex searchsorted

### LINKED ISSUE #14833: MultiIndex searchsorted fails
#### Code Sample, a copy-pastable example if possible

```python
pandas.MultiIndex([[0],["a"]], [[0],[0]]).searchsorted((1,"b"))
```
#### Problem description

The entry `(1,"b")` should come after the existing `(0,"a")` in the `MultiIndex`. (Alternatively, MultiIndex could throw a clean error message.) Instead, an intransparent exception is raised:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.5/site-packages/pandas/core/base.py", line 1156, in searchsorted
    return self.values.searchsorted(key, side=side, sorter=sorter)
TypeError: unorderable types: tuple() > str()
```

This is because `Index.searchsorted` naïvely passes its arguments to `numpy.searchsorted`, which is unaware that its second argument is a sequence of tuples, not a plain array just of dimension one higher.

#### Expected Output
1

#### Output of ``pd.show_versions()``

<details>
# Paste the output here pd.show_versions() here
NSTALLED VERSIONS
------------------
commit: None
python: 3.5.2.final.0
python-bits: 64
OS: Linux
OS-release: 4.8.10-1-ARCH
machine: x86_64
processor: 
byteorder: little
LC_ALL: None
LANG: en_GB.UTF-8
LOCALE: en_GB.UTF-8

pandas: 0.19.1
nose: 1.3.7
pip: 9.0.1
setuptools: 28.8.0
Cython: 0.25.1
numpy: 1.11.2
scipy: None
statsmodels: None
xarray: None
IPython: None
sphinx: None
patsy: None
dateutil: 2.6.0
pytz: 2016.7
blosc: None
bottleneck: None
tables: None
numexpr: None
matplotlib: 1.5.3
openpyxl: None
xlrd: 1.0.0
xlwt: 0.8.0
xlsxwriter: None
lxml: None
bs4: 4.5.1
html5lib: 0.9999999
httplib2: 0.9.2
apiclient: None
sqlalchemy: 1.1.4
pymysql: None
psycopg2: None
jinja2: 2.8
boto: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by ShaharBental**
- [ ] closes #14833 
 - [ ] tests added / passed
 - [ ] passes ``git diff upstream/master | flake8 --diff``
 - [ ] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
@ShaharBental on the issue I indicated that this should simply raise a ``NotImplementedError`` (or maybe a ``ValueError``). I don't think this is actually useful as the indexing methods on a ``MultiIndex`` already do this (they use ``searchsorted`` internally in certain cases).

**Comment by ShaharBental** [comment]
@jreback Sorry, changed it to simply raise a NotImplementedError
I think it should be NotImplementedError and not ValueError, as the problem lies with the type of self and not any specific value.

File: pandas/indexes/multi.py [hunk 1]
```diff
@@ -1470,6 +1470,9 @@ def get_indexer(self, target, method=None, limit=None, tolerance=None):
 
         return _ensure_platform_int(indexer)
 
+    def searchsorted(self, value, side='left', sorter=None):
+        raise NotImplementedError()
```
Comment (jreback) [inline pandas/indexes/multi.py]: add a helpful message that explains one should use multi-index indexing here.

File: pandas/tests/test_multilevel.py [hunk 2]
```diff
@@ -2448,6 +2448,13 @@ def test_iloc_mi(self):
 
         assert_frame_equal(result, expected)
 
+    def test_searchsorted(self):
+        # GH 14833
+        # Test if MultiIndex searchsorted raises the NotImplementedException
+        df_mi = pd.MultiIndex([[0], ["a"]], [[0], [0]])
+        with self.assertRaises(Exception) as context:
+            df_mi.searchsorted((1, "b"))
```
Comment (jreback) [inline pandas/tests/test_multilevel.py]:
```
with self.assertRaises(NotImplementedError):
   ...
```
is enough

**Comment by jreback** [comment]
pls add a note in other api changes in 0.20.0

**Comment by jreback** [comment]
closing as stale. please ping if you'd like to continue.

**Comment by mien** [comment]
is any help required for this PR?

**Pull request closed without merging**